notebook/static/notebook/js/actions.js

```diff
@@ -423,6 +415,21 @@ define(function(require){
                 env.notebook.show_command_palette();
             }
         },
+        'toggle-toolbar':{
+            help: 'hide/show the toolbar',
+            handler : function(env){
+                $('div#maintoolbar').toggle();
```

Comment:
I would but the toolbar isn't in the env :(

This action stuff is a huge mess - I am not going to dive into that and clean everything up...